src/ResourceManager/Compute/Commands.Compute/Strategies/AsyncCmdletExtensions.cs

```diff
@@ -79,20 +81,39 @@ public static void WriteVerbose(this IAsyncCmdlet cmdlet, string message, params
                 });
         }
 
-        sealed class AsyncCmdlet : IAsyncCmdlet
+        sealed class AsyncCmdlet<T> : IAsyncCmdlet
+            where T : PSCmdlet
         {
             public SyncTaskScheduler Scheduler { get; } = new SyncTaskScheduler();
 
-            readonly Cmdlet _Cmdlet;
+            readonly T _Cmdlet;
 
             public List<ITaskProgress> TaskProgressList { get; } 
                 = new List<ITaskProgress>();
 
-            public AsyncCmdlet(Cmdlet cmdlet)
+            public AsyncCmdlet(T cmdlet)
             {
                 _Cmdlet = cmdlet;
             }
 
+            public bool WhatIf
+                => _Cmdlet.MyInvocation.BoundParameters.ContainsKey("WhatIf");
+
+            public IEnumerable<KeyValuePair<string, object>> Parameters
+            {
+                get
+                {
+                    var psName = _Cmdlet.ParameterSetName;
+                    return typeof(T)
+                        .GetProperties()
+                        .Where(p => p
+                            .GetCustomAttributes(false)
+                            .OfType<ParameterAttribute>()
+                            .Any(a => a.ParameterSetName == psName))
```

Comment:
what about parameters like : https://github.com/praries880/azure-powershell/blob/dependency_fix/src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/GetAzureVMCommand.cs#L64
which do not have the property specified? what happens in this case?

Reply:
I've a fix. I'm going to push it soon.